packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-polygon-settings.tsx

```diff
@@ -55,34 +60,87 @@ const LockedPolygonSettings = (props: Props) => {
     } = props;
 
     function handleColorChange(newValue: LockedFigureColor) {
-        onChangeProps({color: newValue});
+        const newProps: Partial<LockedPolygonType> = {
+            color: newValue,
+        };
+
+        // Update the color of the all labels to match the point
+        newProps.labels = labels?.map((label) => ({
+            ...label,
+            color: newValue,
+        }));
+
+        onChangeProps(newProps);
     }
 
     function handlePolygonMove(movement: "up" | "down" | "left" | "right") {
         switch (movement) {
             case "up":
                 onChangeProps({
                     points: points.map(([x, y]) => [x, y + 1]),
+                    labels: labels?.map((label) => ({
+                        ...label,
+                        coord: [label.coord[0], label.coord[1] + 1],
+                    })),
                 });
                 break;
             case "down":
                 onChangeProps({
                     points: points.map(([x, y]) => [x, y - 1]),
+                    labels: labels?.map((label) => ({
+                        ...label,
+                        coord: [label.coord[0], label.coord[1] - 1],
+                    })),
                 });
                 break;
             case "left":
                 onChangeProps({
                     points: points.map(([x, y]) => [x - 1, y]),
+                    labels: labels?.map((label) => ({
+                        ...label,
+                        coord: [label.coord[0] - 1, label.coord[1]],
+                    })),
                 });
                 break;
             case "right":
                 onChangeProps({
                     points: points.map(([x, y]) => [x + 1, y]),
+                    labels: labels?.map((label) => ({
+                        ...label,
+                        coord: [label.coord[0] + 1, label.coord[1]],
+                    })),
                 });
                 break;
         }
     }
 
+    function handleLabelChange(
+        updatedLabel: LockedLabelType,
+        labelIndex: number,
+    ) {
+        if (!labels) {
```

Comment:
nit: I'm personally against empty returns when simply wrapping the logic in an if statement would work just as effectively. Highly recommend the following pattern instead.
`
if (label) {
....
}
`

Reply:
Interesting! This might be an "agree to disagree" situation because I'm pretty solidly in opposition. I prefer returning early, because I find that makes the rest of the function easier to read. It generally reduces indentation, and I feel it makes it clear that we don't have to worry about the null case anymore past that point. I think the "return early" pattern is pretty widely used/taught in my (admittedly limited) experience.

Reply:
It's so funny because while the "return early" pattern was widely taught in my experience. The "empty return" was highly discouraged. Let's agree to disagree here 😉